discord/ui/item.py

```diff
@@ -104,22 +119,59 @@ def row(self) -> Optional[int]:
 
     @row.setter
     def row(self, value: Optional[int]) -> None:
+        if self._is_v2():
+            # row is ignored on v2 components
+            return
+
         if value is None:
             self._row = None
         elif 5 > value >= 0:
             self._row = value
         else:
             raise ValueError('row cannot be negative or greater than or equal to 5')
 
+        if self._rendered_row is None:
+            self._rendered_row = value
+
     @property
     def width(self) -> int:
         return 1
 
     @property
     def view(self) -> Optional[V]:
-        """Optional[:class:`View`]: The underlying view for this item."""
+        """Optional[Union[:class:`View`, :class:`LayoutView`]]: The underlying view for this item."""
         return self._view
 
+    @property
+    def id(self) -> Optional[int]:
+        """Optional[:class:`int`]: The ID of this component."""
+        return self._id
+
+    @id.setter
+    def id(self, value: Optional[int]) -> None:
+        self._id = value
+
+    @property
+    def parent(self) -> Optional[Item[V]]:
+        """Optional[:class:`Item`]: This item's parent. If this item is a :class:`View` this will
+        always be ``None``.
+        """
+        return self._parent
+
+    async def _run_checks(self, interaction: Interaction[ClientT]) -> bool:
+        can_run = await self.interaction_check(interaction)
+
+        if can_run and self._parent:
+            can_run = await self._parent._run_checks(interaction)
+
+        return can_run
+
+    def _update_view(self, view) -> None:
+        self._view = view
+
+    def _has_nested(self) -> bool:
+        return False
```

Comment:
This should be called something else.

```suggestion
    def _has_children(self) -> bool:
        return False
```

Though, again, this seems like another weird way of doing duck typing. To get rid of the type ignores you can probably use `TypeGuard` again.

Something similar to:

```py
if TYPE_CHECKING:
    class ContainerItem(Generic[V]):
        def walk_children(self) -> Generator[Item[V], None, None]:
            ...
```

Then you can use `TypeGuard[ContainerItem[V]]` as the return annotation. This should work since `TypeGuard` (luckily) does not do strict narrowing.

Reply:
I could do that, but ``TypeGuard`` enforces the function to have at least one user-provided parameter. So if I simply updated it to:
```python
def _has_children(self) -> TypeGuard[ContainerItem[V]]:
    return False
```
I would have to ``# type: ignore`` the function definition
<details>

```markdown
User-defined type guard functions and methods must have at least one input parameter    Pylance[reportGeneralTypeIssues](https://github.com/microsoft/pylance-release/blob/main/docs/diagnostics/reportGeneralTypeIssues.md)
```

</details>

And I don't think doing something like ``def _has_children(self, _) -> TypeGuard[ContainerItem[V]]`` is something I should add.

This can be solved if I used ``TypeIs`` instead of ``TypeGuard``.

Reply:
Unfortunate limitation. It seems it works as I expect if it's in a `@staticmethod` and is called like `self.item._has_children(self.item)` but that does read a bit goofy.